### PR TITLE
UI look and feel

### PR DESCRIPTION
**Pull request by amy-hoangg**
1. Create more elements and add classes to existing elements to select in CSS file
2. Modify the existing addRow() to append the first <td> of each row to be the row label
3. Create addHeader()
4. Implement ui.css based on Figma design and Js Spreadsheet
5. Modify the InitializeData() in main.ts to call the new functions

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -29,10 +29,17 @@ export default class LightSheet {
   }
 
   initializeData() {
+    // Create header row and add headers
+    const headerData = Array.from(
+      { length: this.options.data[0].length + 1 }, // Adding 1 for the row number column
+      (_, i) => (i === 0 ? "" : String.fromCharCode(64 + i)), // A for the first column
+    );
```
Comment (rafinutshaw) [inline src/main.ts:36]: this wont hold after column name "Z". 

This link might help to do this better: https://stackoverflow.com/a/8241071

File: src/ui/render.ts [hunk 2]
```diff
@@ -18,15 +19,49 @@ export default class UI {
...
+    this.tableHeadDom.appendChild(headerRowDom);
+
+    for (let i = 0; i < headerData.length; i++) {
+      const headerCellDom = document.createElement("td");
+      headerCellDom.textContent = headerData[i];
+      headerRowDom.appendChild(headerCellDom);
+    }
+  }
+
+  addRow(i: number): Element {
```
Comment (rafinutshaw) [inline src/ui/render.ts:55]: change the variable name from i to somethis more readable, maybe rowIndex/rowNumber

File: src/ui/ui.css [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+.light_sheet_table {
+    border-collapse: separate;
+    /*table-layout: fixed;*/
+    white-space:  nowrap;
+    empty-cells: show;
+    border: 0px;
+    background-color: #fff;
+    /*width: 0;*/
```
Comment (rafinutshaw) [inline src/ui/ui.css:23]: Kindly remove the comment lines

File: src/ui/ui.css [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    
+    border-top: 1px solid transparent;
+    border-left: 1px solid transparent;
+    border-right: 1px solid #ccc;
+    border-bottom: 1px solid #ccc;
+}
```
[the remark this answers is not in the document]
Comment (rafinutshaw) [reply src/ui/ui.css:29]: Actually on top of the same file define your color variables like this

:root {
    --font-size: 15px;

    --blue: #006ac4;
    --blue-300: #3e89ff;
    --blue-500: #73859e;
    --blue-600: #475b79;
}

And then use them elsewhere like this.

.exampleclass {
  border: 1px solid var(--blue-300);
  }

**Review by rafinutshaw — CHANGES_REQUESTED** [review]

File: src/main.ts (hunk 1 above)
Comment (amy-hoangg) [reply src/main.ts:36]: new code is
    // Create header row and add headers
    const headerData = Array.from(
      { length: this.options.data[0].length + 1 }, // Adding 1 for the row number column
      (_, i) => (i === 0 ? "" : this.generateRowLabel(i)), // Generating row labels
    );
    this.ui.addHeader(headerData);


  generateRowLabel(rowIndex: number) {
    let label = "";
    const alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    while (rowIndex > 0) {
      rowIndex--; // Adjust index to start from 0
      label = alphabet[rowIndex % 26] + label;
      rowIndex = Math.floor(rowIndex / 26);
    }
    return label || "A"; // Return "A" if index is 0
  }

File: src/ui/render.ts (hunk 2 above)
Comment (amy-hoangg) [reply src/ui/render.ts:55]: I changed "i" to "rowLabelNumber"

File: src/ui/ui.css (hunk 3 above)
Comment (amy-hoangg) [reply src/ui/ui.css:23]: I have resolved.

File: src/ui/ui.css (hunk 4 above)
[the remark this answers is not in the document]
Comment (rafinutshaw) [reply src/ui/ui.css]: You are on the right track but my instrucitions may not have been clear.
I ment to say make variables out of every color used in the css file. and then replace the colors with the variable. 

.exampleclass {
//do this 
border: 1px solid var(--blue-300);
//not this
  border: var(--border-top);
}

the purpose is to make the colors variables resusable anywhere else is the app. without the need to looking into specific hexcodes everytime. and if we want to change the color theme of the app we just change the color variables

[the remark this answers is not in the document]
Comment (rafinutshaw) [reply src/ui/ui.css]: kindly add a prefix to the variables names as well.
example: --lightsheet-blue-400

[the remark this answers is not in the document]
Comment (rafinutshaw) [reply src/ui/ui.css]: @amy-hoangg

[the remark this answers is not in the document]
Comment (rafinutshaw) [reply src/ui/ui.css]: Hi I had some time so I have made the necessary changes. You can have a look into it and merge

**Review by rafinutshaw — APPROVED** [review]
Hi I had some time so I have made the necessary changes. You can have a look into it and merge.

**Pull request merged**